src/librustdoc/formats/item_type.rs

```diff
@@ -175,6 +179,8 @@ impl ItemType {
             ItemType::ProcAttribute => "attr",
             ItemType::ProcDerive => "derive",
             ItemType::TraitAlias => "traitalias",
+            ItemType::TyAssocType => "tyassociatedtype",
```

Comment:
About this, it's what I said: instead of creating new anchors, you could reuse `associatedtype` and `associatedconstant`. It would allow to not break any existing links to current documentation. It would also remove the need for the new categories in JS. What do you think @rust-lang/rustdoc ?

Reply:
> About this, it's what I said

Yeah, gotcha. I didn't change it (yet) since Manishearth was happy / fine with it and you were about okay with it. I will wait what the rest of the team has to say. :+1:

Reply:
Yes, please don't touch anything until they agree. :smile:

Reply:
I would really like know why the distinction exists at all. If we know what the rationale was, we can figure out if it also applies here.

As near as I can tell, this distinction can be traced back about as far as anything can be traced back: [it was part of the very first commit adding an HTML backend to rustdoc_ng](https://github.com/rust-lang/rust/blob/4fd061c426902b0904c65e64a3780b21f9ab3afb/src/rustdoc_ng/html/render.rs#L516-L532), and no reason seems to have been given for carrying it over into the HTML and the search index.

Is any part of the rustdoc frontend using this information? Or can it safely be removed?

Reply:
I'd be in favor of removing this distinction (and the newly added ones) as I don't really see their point...

Reply:
So I think we all agree that this distinction shouldn't exist. Please reuse the existing CSS classes then.

Reply:
> Please reuse the existing CSS classes then.

Will do. Should I keep the changes made to `clean::Item`? They reduce the size as you wrote. On the other hand, they introduce the “weird” `Ty` prefix to more places in `rustdoc`.

@rustbot author

Reply:
Will do!